pkg/hcl2nix/lock.go

```diff
@@ -226,5 +227,23 @@ func mapPackageCategory(packages Packages) map[string][]Category {
 	addCategory(packages.Runtime, Runtime)
 	addCategory(packages.Development, Development)
 
-	return m
+	// Sort keys alphabetically
+	sortedKeys := sortMapKeys(m)
+
+	// Create a new map with sorted keys
+	sortedMap := make(map[string][]Category)
+	for _, key := range sortedKeys {
+		sortedMap[key] = m[key]
+	}
+
+	return sortedMap
+}
+
+func sortMapKeys(m map[string][]Category) []string {
```

Comment:
I don't think maps ordering is reliable in Go(even if it looks like it works sometimes). It's a victim of the infamous [Hyrum's Law](https://www.hyrumslaw.com/)
  
I think what we should do instead of sort the slices alphabetically every time(once we get a slice back from map, etc).  Does that make sense?

https://groups.google.com/g/golang-nuts/c/TDwGcRQe6mQ?pli=1

Reply:
Thankyou for the review ! I will try to make the needed changes.